packages/ra-ui-materialui/src/layout/TopToolbar.tsx

```diff
@@ -46,3 +46,5 @@ const StyledToolbar = styled(Toolbar, {
         paddingBottom: 0,
     },
 }));
+
+const sanitizeToolbarRestProps = ({ hasCreate, ...props }: any) => props;
```

Comment:
You should probably add the others as well, `hasList`, etc

Reply:
`hasCreate` is the only superfluous prop injected by [`ListView`](https://github.com/marmelab/react-admin/blob/master/packages/ra-ui-materialui/src/list/ListView.tsx#L60) (through [`ListToolbar`](https://github.com/marmelab/react-admin/blob/master/packages/ra-ui-materialui/src/list/ListToolbar.tsx#L21)).